modules/navier_stokes/src/functormaterials/LinearFrictionFactorFunctorMaterial.C

```diff
@@ -0,0 +1,84 @@
+
+  params.addRequiredParam<MooseFunctorName>("functor_name",
+                                            "The name of functor storing the friction factor");
+  params.addRequiredParam<MooseFunctorName>(NS::porosity, "porosity");
+  params.addParam<MooseFunctorName>(
+      NS::superficial_velocity_x, 0, "The x component of the fluid superficial velocity variable.");
+  params.addParam<MooseFunctorName>(
+      NS::superficial_velocity_y, 0, "The y component of the fluid superficial velocity variable.");
+  params.addParam<MooseFunctorName>(
+      NS::superficial_velocity_z, 0, "The z component of the fluid superficial velocity variable.");
+  params.addParam<RealVectorValue>(
+      "A", RealVectorValue(1, 1, 1), "Coefficient of the A * f(t) term");
+  params.addParam<RealVectorValue>(
+      "B", RealVectorValue(1, 1, 1), "Coefficient of the B * g(t) * |v_I| term");
+  params.addParam<MooseFunctorName>("f", "Functor f in the A * f(t) term");
+  params.addParam<MooseFunctorName>("g", "Functor g in the B * g(t) * |v_I| term");
+  return params;
+}
+
+LinearFrictionFactorFunctorMaterial::LinearFrictionFactorFunctorMaterial(
+    const InputParameters & parameters)
+  : FunctorMaterial(parameters),
+    _functor_name(getParam<MooseFunctorName>("functor_name")),
+    _A(getParam<RealVectorValue>("A")),
+    _B(getParam<RealVectorValue>("B")),
+    _f(getFunctor<ADReal>("f")),
+    _g(getFunctor<ADReal>("g")),
+    _eps(getFunctor<ADReal>(NS::porosity)),
+    _superficial_vel_x(getFunctor<ADReal>(NS::superficial_velocity_x)),
+    _superficial_vel_y(getFunctor<ADReal>(NS::superficial_velocity_y)),
+    _superficial_vel_z(getFunctor<ADReal>(NS::superficial_velocity_z))
+{
+  // Check dimension of the mesh
+  const unsigned int num_components_specified =
+      parameters.isParamSetByUser(NS::superficial_velocity_x) +
+      parameters.isParamSetByUser(NS::superficial_velocity_y) +
+      parameters.isParamSetByUser(NS::superficial_velocity_z);
+  if (num_components_specified != blocksMaxDimension())
+    mooseError("Only ",
+               num_components_specified,
+               " superficial velocity components were provided for a mesh of dimension ",
+               blocksMaxDimension());
+
+  addFunctorProperty<ADRealVectorValue>(
+      _functor_name,
+      [this](const auto & r, const auto & t) -> ADRealVectorValue
+      {
+        // Compute speed
+        // see PINSFVSpeedFunctorMaterial.C for explanation
+        ADReal speed;
+        if ((MooseUtils::absoluteFuzzyEqual(_superficial_vel_x(r, t), 0)) &&
+            (MooseUtils::absoluteFuzzyEqual(_superficial_vel_y(r, t), 0)) &&
+            (MooseUtils::absoluteFuzzyEqual(_superficial_vel_z(r, t), 0)))
+          speed = 1e-42;
+
+        speed = ADRealVectorValue(
+                    _superficial_vel_x(r, t), _superficial_vel_y(r, t), _superficial_vel_z(r, t))
+                    .norm() /
+                _eps(r, t);
```

Comment:
I would save off a local velocity variable and then call `NS::computeSpeed`
```suggestion
        const ADRealVectorValue superficial_vel(_superficial_vel_x(r, t),
                                                _superficial_vel_y(r, t),
                                                _superficial_vel_z(r, t));
        const auto speed = NS::computeSpeed(superficial_vel) / _eps(r, t);
```